src/vmm/src/vmm_config/hotplug.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::io;
+
+use serde::{Deserialize, Serialize};
+
+use crate::vstate::vcpu::VcpuError;
+use crate::StartVcpusError;
+
+/// Errors associated with hot-plugging vCPUs
+#[derive(Debug, thiserror::Error, displaydoc::Display)]
+pub enum HotplugVcpuError {
+    /// The number of vCPUs added must be greater than 0.
+    VcpuCountTooLow,
+    /// The number of vCPUs added must be less than 32.
+    VcpuCountTooHigh,
+    /// Event fd error: {0}
+    EventFd(io::Error),
+    /// Error creating the vcpu: {0}
+    VcpuCreate(VcpuError),
+    /// Failed to start vCPUs
+    VcpuStart(StartVcpusError),
+}
+
+/// Config for hotplugging vCPUS
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct HotplugVcpuConfig {
+    /// Number of vcpus to start.
+    pub vcpu_count: u8,
+}
```

Comment:
If we will have 1 endpoint for all hot plugging I would prefer to have 1 enum that represents all options. Smth like this:
```rust
enum HotplugRequest {
  Vcpu(HotplugVcpu),
  // Memory(HotplugMemory),
}
```
This better represents available options. Additionally way we don't need to do any additional checks like you do in the `request/hotplug.rs`.

Reply:
As you suggested, I've added enums above the vCPU specific ones in order to unify vCPU hotplugging with other resources (should they be implemented). Let me know if it looks okay

Reply:
I don't see any changes

Reply:
Can you clarify what you were looking for? I've left the existing enums as is, but have added an extra enum above `HotplugRequestConfig`, which would unify the two different types of request into one enum. The result of this is that there is a single VmmAction for hotplugging, `VmmAction::HotplugRequest(HotplugRequestConfig)`, and at this point the handling of the different hotplug requests is handled.

Reply:
Ok, I see this now. For some reason if I look at the commits 1 by 1 I don't see the change, but if I look at all changed files it is there.